### DISCUSSION TITLE
Function 'onChangePhoneNumber' is called with empty value every time component is rendered even with defaultValue

### DISCUSSION THREAD

**Discussion by victorgoecking** [AstrOOnauta/react-native-international-phone-number#24] [Q&A]
I'm using "React-Hook-Form + Typescript + Default Phone Number Value". I receive the api data already passed to the hook-form's defaultValue and also passing the "value" and "defaultValue="+5500999999999" in the <PhoneInput />, but when the component is rendered it calls the "onChangePhoneNumber" function defining the value as empty.

I'm using a small strategy to check if it's the first rendering so as not to set an empty value.
![image](https://github.com/AstrOOnauta/react-native-international-phone-number/assets/9842243/2481bf96-7952-41ab-92e4-87f402da0d2b)

What could I be doing wrong?
![image](https://github.com/AstrOOnauta/react-native-international-phone-number/assets/9842243/22d56853-d4fd-404a-a4ca-7727b33e3606)

**Comment by AstrOOnauta** [comment]
hey @victorgoecking

I tried to reproduce your situation using the [example](https://github.com/AstrOOnauta/react-native-international-phone-number#react-hook-form--typescript--default-phone-number-value) on README with React-Hook-Form + Typescript + Default Phone Number Value and the input value dont stay empty after rerender...

Can you deploy this code anywhere? Maybe, I (or another) can help you...

**Reply by victorgoecking** [reply]
hey @AstrOOnauta 

I reproduced my situation using the [example](https://github.com/AstrOOnauta/react-native-international-phone-number#react-hook-form--typescript--default-phone-number-value) on README and actually sets it correctly.

But in it the value coming from the controller is not defined, in a use case for editing this value will come in the controller's render and thus causing onChangePhoneNumber to execute with an empty value.

![image](https://github.com/AstrOOnauta/react-native-international-phone-number/assets/9842243/c6b07368-368e-46fb-a28c-80901fa87855)
